src/AddImageChecks.cpp

```diff
@@ -259,6 +259,11 @@ Stmt add_image_checks(Stmt s,
                            << j << "\n";
             }
 
+            Expr negative_extent_condition = actual_extent >= 0;
+            Expr negative_extent_error = Call::make(Int(32), "halide_error_buffer_extents_negative",
+                                                    {error_name, j, actual_extent}, Call::Extern);
+            asserts_required.push_back(AssertStmt::make(negative_extent_condition, negative_extent_error));
```

Comment:
True: and doing that move reveals that my testcase triggers the pre-existing OOB checks. Let me look more closely at the original failure case I saw and see if I can just tweak the OOB check to notice it as well.